common/TruffleConfig.js

```diff
@@ -47,15 +49,12 @@ function addPublicNetwork(networks, name, networkId) {
     gasPrice: gasPx
   };
 
-  // If a custom node URL is provided, use that. Otherwise use an infura websocket connection.
-  const nodeUrl = customNodeUrl || `wss://${name}.infura.io/ws/v3/${infuraApiKey}`;
-
   // GCS ManagedSecretProvider network.
-  networks[name] = {
+  networks[name + "_gckms"] = {
     ...options,
-    provider: function() {
+    provider: function(provider = nodeUrl) {
```

Comment:
This modification lets us re-use the provider function for a given network name while overriding the `provider` prop. If this is not provided this is the original URL.